### PR TITLE
Feature/persistent metrics

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies

…

## This introduces a breaking change

- [ ] Yes
- [ ] No

## This Pull Request implements

… Explain your changes.

### CODE REVIEW

File: app.sh [hunk 1]
```diff
@@ -8,16 +8,16 @@ if [ "$SUBCOMMAND" = "producer" ]
 then
```
Comment (pacospace) [inline app.sh:8]: what about making it like https://github.com/thoth-station/investigator/blob/master/app.sh?

File: consumer.py [hunk 2]
```diff
@@ -21,7 +21,8 @@
 from thoth.python import Source
 from thoth.common import init_logging
 from thoth.messaging import MissingPackageMessage, MissingVersionMessage, HashMismatchMessage, MessageBase
-from .process_message import process_mismatch, process_missing_package, process_missing_version
+from package_update.process_message import process_mismatch, process_missing_package, process_missing_version
+# from package_update import __service_version__
```
Comment (pacospace) [inline consumer.py:25]: leftover?

File: consumer.py [hunk 3]
```diff
@@ -34,55 +35,40 @@
 
 init_logging()
 
-_LOGGER = logging.getLogger("thoth.package_update")
+_LOGGER = logging.getLogger(__name__)
+# _LOGGER.info("Thoth Package Update consumer v%s", __service_version__)
```
Comment (pacospace) [inline consumer.py:39]: leftover?

File: consumer.py [hunk 4]
```diff
@@ -34,55 +35,40 @@
...
+@app.page("/metrics")
+async def get_metrics(self, request):
+    """Serve the metrics from the consumer registry."""
+    return web.Response(text=generate_latest().decode("utf-8"))
+
 
+@app.page("/_health")
+async def get_health(self, request):
+    """Serve a readiness/liveness probe endpoint."""
+    data = {"status": "ready", "version": __package_update_version__}
```
Comment (pacospace) [inline consumer.py:56]: `__service_version__`?

File: package_update/metrics.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
+
+#!/usr/bin/env python3
+# thoth-investigator
```
Comment (pacospace) [inline package_update/metrics.py:3]: copy/paste

File: package_update/metrics.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+
+"""This is Thoth investigator consumer metrics."""
```
Comment (pacospace) [inline package_update/metrics.py:20]: copy/paste?

File: package_update/metrics.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+
+from package_update import __service_version__
+
+from prometheus_client import Gauge, Counter
+
+
+# add the application version info metric
+investigator_info = Gauge("package_update_consumer_info", "Package Update Version Info", labelnames=["version"])
+investigator_info.labels(version=__service_version__).inc()
```
Comment (pacospace) [inline package_update/metrics.py:30]: variables name?

File: package_update/metrics.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+# Metrics for Kafka
+mismatch_in_progress = Gauge(
+    "hash_mismatch_in_progress",
+    "Total number of hashmismatch messages currently being processed. This will be 1 or 0 unless an async version is implemented",
```
Comment (pacospace) [inline package_update/metrics.py:35]: `hash mismatch`

File: package_update/metrics.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+)
+mismatch_exceptions = Counter(
```
Comment (pacospace) [inline package_update/metrics.py:37]: what about adding label `message_type` to metrics? so we don't need to create one for each message?

File: consumer.py (hunk 2 above)
Comment (pacospace) [reply consumer.py:25]: `from thoth.package_update import __version__ as __package_update_version__`

File: producer.py [hunk 10]
```diff
@@ -30,19 +30,17 @@
...
 
+from thoth.package_update import __service_version__ as __package_update_version__
+
 init_logging()
 
-_LOGGER = logging.getLogger("thoth.package_update")
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.info("Thoth package update producer v%s", __package_update_version__)
 
 app = MessageBase.app
```
Comment (pacospace) [inline producer.py:42]: `app = MessageBase().app`?

Comment (KPostOffice) [reply producer.py:42]: Yeah and I'll update `thoth.messaging` in Pipfile.lock

Comment (pacospace) [reply producer.py:42]: https://github.com/thoth-station/package-update-job/pull/108

**Comment by pacospace** [comment]
please rebase @KPostOffice

**Pull request merged**